.changeset/cruel-colts-smoke.md

```diff
@@ -0,0 +1,33 @@
+---
+"@pnpm/config": major
+"pnpm": major
+---
+
+Replace workspace project specific `.npmrc` with `projectSettings` in `pnpm-workspace.yaml`.
+
+A workspace manifest with `projectSettings` would look something like this:
+
+```yaml
+# File: pnpm-workspace.yaml
+packages:
+  - 'packages/*'
+projectSettings:
+  'project-1':
+    saveExact: true
+  'project-2':
+    savePrefix: '~'
+```
+
+Or this:
+
+```yaml
+# File: pnpm-workspace.yaml
+packages:
+  - 'packages/*'
+projectSettings:
+  - match:
+      - 'project-1'
+      - 'project-2'
+    settings:
+      saveExact: true
+```
```

Comment:
I am not sure we need two separate syntaxes. After all only a few settings are supported. I am not sure it is or will be a frequently used feature.

The field should probably be renamed to `packageConfigs` to follow the convention already used in the `pnpm-workspace.yaml` file (via the `packages` field).

The second way of declaring it can be made more compact by flattening the fields like 

```yaml
packageConfigs:
  - match:
      - 'project-1'
      - 'project-2'
    saveExact: true
```

Reply:
ok, just rename it.

Reply:
we have to use packageConfigs to adhere the already used term "packages"